### PR TITLE
Airport Extension

### PR DESCRIPTION
**Pull request by siddharthsharma94**
## Description

View[ Airport Apps](https://app.airport.community) directly from Raycast!

## Type of change

- New extension

## Screenshot

Airport-Raycast

## Checklist

- [x] I read the [contribution guidelines](contributing.md)

### CODE REVIEW

File: extensions/airport/package.json [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+  "description": "Airport",
+  "icon": "logo2.png",
+  "author": "siddharthsharma94",
+  "license": "MIT",
+  "commands": [
+    {
+      "name": "index",
+      "title": "Apps",
+      "subtitle": "See Airport Apps",
+      "description": "Airport is the best place to discover new TestFlight apps from developers.",
```
Comment (thomaspaulmann) [inline extensions/airport/package.json:13]: Discover apps from developers on Airport.

File: extensions/airport/src/index.tsx [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+        waitlist: appsResponse?.apps.all.filter(
+          (app) =>
+            !appsResponse?.apps.popular.includes(app.id) &&
+            !appsResponse?.apps.featured.includes(app.id) &&
+            !appsResponse?.apps.new.includes(app.id) &&
+            !appsResponse?.apps.upcoming.includes(app.id) &&
+            appsResponse?.apps.waitlist.includes(app.id)
+        ),
+      }));
+      showToast(ToastStyle.Success, "Fetched Latest Apps");
```
Comment (thomaspaulmann) [inline extensions/airport/src/index.tsx:80]: It's enough to show the loading indicator in the searchbar. You don't need to add the toast. More info [here](https://developers.raycast.com/information/best-practices#show-loading-indicator).

```suggestion
      const appsResponse = await fetchApps();
      setState((oldState) => ({
        ...oldState,
        all: appsResponse?.apps.all.filter(
          (app) =>
            !appsResponse?.apps.popular.includes(app.id) &&
            !appsResponse?.apps.featured.includes(app.id) &&
            !appsResponse?.apps.new.includes(app.id) &&
            !appsResponse?.apps.upcoming.includes(app.id) &&
            !appsResponse?.apps.waitlist.includes(app.id)
        ),
        popular: appsResponse?.apps.all.filter((app) => appsResponse?.apps.popular.includes(app.id)),
        featured: appsResponse?.apps.all.filter((app) => appsResponse?.apps.featured.includes(app.id)),
        new: appsResponse?.apps.all.filter(
          (app) =>
            !appsResponse?.apps.popular.includes(app.id) &&
            !appsResponse?.apps.featured.includes(app.id) &&
            appsResponse?.apps.new.includes(app.id) &&
            !appsResponse?.apps.upcoming.includes(app.id) &&
            !appsResponse?.apps.waitlist.includes(app.id)
        ),
        upcoming: appsResponse?.apps.all.filter(
          (app) =>
            !appsResponse?.apps.popular.includes(app.id) &&
            !appsResponse?.apps.featured.includes(app.id) &&
            !appsResponse?.apps.new.includes(app.id) &&
            appsResponse?.apps.upcoming.includes(app.id) &&
            !appsResponse?.apps.waitlist.includes(app.id)
        ),
        waitlist: appsResponse?.apps.all.filter(
          (app) =>
            !appsResponse?.apps.popular.includes(app.id) &&
            !appsResponse?.apps.featured.includes(app.id) &&
            !appsResponse?.apps.new.includes(app.id) &&
            !appsResponse?.apps.upcoming.includes(app.id) &&
            appsResponse?.apps.waitlist.includes(app.id)
        ),
      }));
```

**Review by thomaspaulmann — CHANGES_REQUESTED** [review]
Hey @siddharthsharma94,

this is a pretty cool extension! I love bringing Airport and Raycast together. I left a few minor things to tidy up the extension inline. Then it should be good to go. Can't wait to see folks discovering beta software from other developers right in Raycast!!!

**Comment by siddharthsharma94** [comment]
Just added your requested changes! @thomaspaulmann

File: extensions/airport/package.json [hunk 3]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "airport",
-  "title": "Airport",
+  "title": "Airport - Discover TestFlight Apps",
```
Comment (thomaspaulmann) [inline extensions/airport/package.json:3]: Really nice title!

File: extensions/airport/src/index.tsx [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+  return (
+    <List.Item
+      id={`${type}-${app.id}`}
+      key={`${type}-${app.id}`}
+      title={app.name}
+      subtitle={app.description}
+      icon={{ source: app.icon }}
+      actions={
+        <ActionPanel>
+          <OpenInBrowserAction url={`https://app.airport.community/app/${app.id}?utm_source=raycast`} />
```
Comment (thomaspaulmann) [inline extensions/airport/src/index.tsx:131]: Do you really need this UTM parameter? We don't do this to respect the user's privacy. I know that this isn't user related but some folks don't like UTM either. Wanna remove it?

**Review by thomaspaulmann — APPROVED** [review]
Thanks for making the changes so quickly. The title looks very nice now. I left one comment about the UTM parameter. Do you want to remove it before I merge your extension?

**Comment by grzegorzkrukowski** [comment]
Hey @siddharthsharma94 we just found out that icon in your extension has to be increased - we changed requirement to be 512 x 512 pixels.
Also in `assets` folder there is unused command-icon.png file.

Do you mind updating the icon to correct 512 x 512 size and removing unused one ?

Thanks in advance.

**Comment by siddharthsharma94** [comment]
Sounds good, I’ll make a PR to change the icon later today!

**Comment by siddharthsharma94** [comment]
@grzegorzkrukowski just made a PR [here](https://github.com/raycast/extensions/pull/323)

**Pull request merged**